src/compiler/sys.ts

```diff
@@ -221,7 +228,7 @@ namespace ts {
 
             // average async stat takes about 30 microseconds
             // set chunk size to do 30 files in < 1 millisecond
-            function createWatchedFileSet(interval = 2500, chunkSize = 30) {
+            function createPollingWatchedFileSet(interval = 2500, chunkSize = 30) {

```

Comment:
Our polling mechanism is different from `fs.watchFile`. `fs.watchFile` polls each file independently, so if you got many files to watch, it might increase the work load a lot. In our approach, we poll a fixed number of files (`chunkSize`) each number, so if the `chunkSize` is 30 and you have 90 files to watch, the work load is still the same for each polling, only that changes will be detected after 3 interval time instead of just 1. This way we keep the polling work load consistent